2-mandatory/9-cheap-diner.js

```diff
@@ -19,7 +19,21 @@ chosenMeal(emptyArray)
 Should give the answer "Nothing :("
 **/
 
-function chooseMeal(mealArray) {}
+function chooseMeal(mealArray) {
+  let selectedMeal = 0;
+  if (mealArray.length === 0) selectedMeal = "Nothing :(";
+  else if (mealArray.length === 1) selectedMeal = mealArray[0].name;
+  else {
+    const allPrices = mealArray.map((meal) => meal["price"]);
+    const secondMinimum = allPrices.sort((a, b) => a - b)[1];
```

Comment:
```suggestion
 mealArray.sort((a, b) => a.price - b.price);
```
This will sort by price directly without using the map (it's included on sort) and will change the mealArray directly